proposals/solid-notifications.md

```diff
@@ -0,0 +1,270 @@
+    "rate": "PT10s"
+}
+```
+
+A successful response will contain a URL that can be used directly with a JavaScript client
+
+```http
+HTTP/2
+Content-Type: application/ld+json
+
+{
+    "@context": "https://www.w3.org/ns/solid/notification/v1",
+    "type": "WebSocketSubscription2021",
+    "endpoint": "wss://websocket.example/?auth=Ys3KiUq"
+}
+```
+
+In JavaScript, a client can use the data in the response to establish a connection to the WebSocket endpoint.
+
+```javascript
+const ws = new WebSocket(endpoint, type)
+```
+
+A client will define how to handle events, especially the onclose and onmessage events.
+
+```javascript
+ws.onclose(evt => reconnect(evt));
+ws.onmessage(evt => console.log("Message received: ", evt))
+```
+
+## Authentication and Authorization
+
+The Subscription API requires authorization. This document does not define the specific technology used to authorize requests; rather, it defers to the Solid Protocol sections on Authentication and Authorization. It is out of scope for this document to describe how a client fetches an authorization token. Solid-OIDC is one example of an authentication mechanism that could be used with Solid Notifications.
+
+Resource-based Access controls MUST be enforced for every subscription. A client MUST be permitted to read the resource to which it subscribes.
+
+## Notification Data Model
+
+The content of a Solid Notification is defined in terms of a W3C Activity Streams 2.0 document. As such, JSON-LD 1.0 is a required baseline serialization format, though other formats are not prohibited. JSON-LD version 1.1 is recommended when using JSON-LD serializations.
+
+```javascript
+{
+   "@context":[
+      "https://www.w3.org/ns/activitystreams",
+      "https://www.w3.org/ns/solid/notification/v1"
+   ],
+   "id":"urn:uuid:<uuid>",
+   "type":[
+     "Update"
+   ],
+   "actor":[
+      "<WebID>"
+   ],
+   "object":{
+      "id":"https://pod.inrupt.com/<user>/some-container/",
+      "type":[
+         "http://www.w3.org/ns/ldp#BasicContainer",
+         "http://www.w3.org/ns/ldp#Resource",
+         "http://example.com/SomeType"
+      ]
+   },
+   "state": "1234-5678-90ab-cdef-12345678",
+   "published":"2021-08-05T01:01:49.550044Z"
+}
+```
+
+**Note**: the Solid JSON-LD context resource does not currently exist. An implementation may choose to augment the context definition.
+
+The content of these serialized notifications may be extended, but the example here provides a minimal baseline that clients can expect.
+
+## Subscription Types
+
+This document defines four initial subscription types:
+
+* `WebSocketSubscription2021`
+* `EventSourceSubscription2021`
+* `LinkedDataNotificationSubscription2021`
+* `WebHookSubscription2021`
+
+**Note**: the naming convention used here follows a pattern used by the Verifiable Credentials community, making it possible to distinguish between different versions of these subscription types. Other naming conventions can achieve the same end.
+
```

Comment:
To hide some of the complexity, would there be any advantage of having a universal notifications layer that will handle common elements of the different subscription types (similar in idea as https://github.com/solid/authorization-panel/issues/253 for authorization)?

Reply:
#11